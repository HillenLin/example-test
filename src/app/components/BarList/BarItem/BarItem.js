import React from 'react'
import Styled from 'styled-components'

const StyledBar = Styled.div`
  width: 80%;
  height: 30px;
  position: relative;
  background-color: #e0e0e0;
  margin: 15px auto;
  padding: 0;
  border-radius: 8px;
`
const StyledSpan = Styled.span`
display: block;
height: 100%;
width: ${props => (props.progress >= 100 ? '100%' : props.progress + '%')};
background-color: ${props => (props.progress >= 100 ? '#ff3d00' : '#5393ff')};
position: relative;
text-align: center;
overflow: hidden;
border-radius: 8px;
&:after{
  content: "";
  position: absolute;
  top: 0; left: 0; bottom: 0; right: 0;
  z-index: 1;
  overflow: hidden;
}
`

const StyledText = Styled.p`
  position: absolute;
  z-index: 3;
  top: -12px;
  right: 40%;
`
const BarItem = ({ progress }) => {
  return (
    <li>
      <StyledBar>
        <StyledText>{progress + '%'}</StyledText>
        <StyledSpan progress={progress} />
      </StyledBar>
    </li>
  )
}

export default BarItem
