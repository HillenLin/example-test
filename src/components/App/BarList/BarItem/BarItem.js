import React from 'react'
import styled from 'styled-components'

const StyledBar = styled.div`
  background-color: #e0e0e0;
  border-radius: 8px;
  border: 1px solid ${props => (props.isSelected ? '#ff3d00' : '#e0e0e0')};
  height: 30px;
  margin: 15px auto;
  padding: 2px;
  position: relative;
  width: 80%;
`
const StyledSpan = styled.span`
  background-color: ${props => (props.progress >= 100 ? '#ff3d00' : '#64b5f6')};
  border-radius: 4px;
  display: block;
  height: 100%;
  overflow: hidden;
  position: relative;
  text-align: center;
  width: ${props => (props.progress >= 100 ? '100%' : props.progress + '%')};
  &:after {
    bottom: 0;
    content: '';
    left: 0;
    overflow: hidden;
    position: absolute;
    right: 0;
    top: 0;
    z-index: 1;
  }
`

const StyledText = styled.p`
  bottom: 0;
  font-family: 'Work Sans', sans-serif;
  font-size: 16px;
  margin: 0;
  position: absolute;
  right: 50%;
  z-index: 2;
`
const BarItem = ({ isSelected, progress }) => {
  return (
    <li>
      <StyledBar isSelected={isSelected}>
        <StyledText>{progress + '%'}</StyledText>
        <StyledSpan progress={progress} />
      </StyledBar>
    </li>
  )
}

export default BarItem
