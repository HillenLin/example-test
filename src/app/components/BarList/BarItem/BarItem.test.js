import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BarItem from './index'

test('test BarItem', () => {
  const wrapper = shallow(<BarItem key={9} progress={90} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
