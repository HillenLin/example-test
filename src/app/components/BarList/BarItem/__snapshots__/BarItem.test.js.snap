// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test BarItem 1`] = `
<li>
  <styled.div>
    <styled.span
      progress={90}
    >
      90%
    </styled.span>
  </styled.div>
</li>
`;
